docs/topics/forms/media.txt

```diff
@@ -271,29 +303,47 @@ Or if :mod:`~django.contrib.staticfiles` is configured using the
     <script src="https://static.example.com/animations.27e20196a850.js"></script>
     <script src="https://othersite.com/actions.js"></script>
 
+.. _form-media-asset-objects:
+
 Paths as objects
 ----------------
 
-Asset paths may also be given as hashable objects implementing an
-``__html__()`` method. The ``__html__()`` method is typically added using the
-:func:`~django.utils.html.html_safe` decorator. The object is responsible for
-outputting the complete HTML ``<script>`` or ``<link>`` tag content:
+Assets may also be object-based, using :class:`.CSS` and :class:`.JS`.
+Furthermore, these allow you to pass custom HTML attributes::
 
-.. code-block:: pycon
+    class Media:
```

Comment:
Asked on the forum, but will ask here as well so folks get notified

Are we all happy with the names of `CSS` and `JS` for these classes?
Not sure if having an all-caps class name is something we should avoid. Maybe `CSSAsset`/`JSAsset`?

Reply:
We might also consider their corresponding HTML tags: `Script` & `Link`

Reply:
I like those! I'm not necessarily against `CSS` and `JS` but I'm conscious it's hard to rename things

Reply:
I don't care much either way but there's a precedent for using `CSS` and `JS`, since those are the keywords used by `forms.Media`. If we were starting from scratch it would maybe be a different story, but I think since we're basically enabling additional use cases for the existing `css` and `js` names using `CSS` and `JS` should be fine as well.

Reply:
> I don't care much either way but there's a precedent for using `CSS` and `JS`, since those are the keywords used by `forms.Media`. If we were starting from scratch it would maybe be a different story, but I think since we're basically enabling additional use cases for the existing `css` and `js` names using `CSS` and `JS` should be fine as well.

You make a valid point. Form media currently revolves around this concept.

Reply:
After some consideration, I would strongly recommend dropping CSS altogether. There isn't a strong use case. Modern web components ship with their own styles. Exposing `link` comes with multiple new challenges, especially since Form.media assets are path-based. A simple preload and stylesheet combo would break the current implementation of form Media. Besides, they dict-key media attribute warning, isn't intuitive either.

Looking into the future, I would envision `Form.media` just being a simple list of object-based assets or maybe split into head and body. But this was meant as a quick win. It's a win for me, if we get `Script` into Django 5.2.

@sarahboyce if, you are in favor, I would split this up.

Reply:
Always happy for less - let's split :grin:

Reply:
It has been done.